src/components/upload/nz-basic-upload.component.ts

```diff
@@ -0,0 +1,149 @@
+import { Component, EventEmitter, Input, Output, OnInit, ElementRef, ViewChild } from '@angular/core';
+import { UidService } from './uid/uid.service';
+import defaultRequest from './request';
+import { attrAccept } from './utils';
+
+@Component({
+    selector: 'nz-basic-upload',
+    providers: [UidService],
+    templateUrl: './nz-basic-upload.component.html'
+})
+export class NzBasicUploadComponent implements OnInit {
+
+    _classMap;
+    _prefixCls = 'ant-upload';
+    _reqs = {};
+
+    @Input() accept: string;
+    @Input() action: string;
+    @Input() autoUpload = true;
+    @Input() beforeUpload: (file: File, files: FileList) => boolean | Promise<any> = null;
+    @Input() customRequest: any;
+    @Input() data: any;
+    @Input() disabled = false;
+    @Input() headers: string | { [name: string]: string | string[] } = {};
+    @Input() multiple = false;
+    @Input() name: string;
+    @Input() withCredentials = false;
+
```

Comment:
currently, ng zorro team the input/output use `nz` prefix. and the output `nz` prefix should replace `on`. so like `onError` should be nzError

Reply:
@LinBoLen  is right, please change the input and output name.

Reply:
`nz` prefix is needed